src/lib.rs

```diff
@@ -0,0 +1,111 @@
+extern crate tokio_trace;
+
+struct Span {
+    label: String,
+    ref_count: usize,
+}
+
+use std::fmt;
+use std::sync::Mutex;
+
+use tokio_trace::field::{Field, Visit};
+use tokio_trace::metadata::Level;
+use tokio_trace::{span, Event, Metadata, Subscriber};
+pub struct Visitor;
+
+impl Visit for Visitor {
+    fn record_debug(&mut self, field: &Field, value: &fmt::Debug) {
+        if field.name() == "message" {
+            print!("{:?}: ", value);
+        } else {
+            print!("{} = {:?} ", field.name(), value);
+        }
+    }
+}
+
+pub struct DemoSubscriber {
+    spans: Mutex<Vec<Span>>,
+    stack: Mutex<Vec<u64>>,
+}
+
+impl DemoSubscriber {
+    pub fn new() -> DemoSubscriber {
+        DemoSubscriber {
+            spans: Mutex::default(),
+            stack: Mutex::default(),
+        }
+    }
+}
+
+impl Subscriber for DemoSubscriber {
+    fn enabled(&self, _metadata: &Metadata) -> bool {
+        true
+    }
+
+    fn new_span(&self, attributes: &span::Attributes) -> span::Id {
+        // TODO: correlate spans based on their metadata
+        let mut spans = self.spans.lock().unwrap();
+        let id = spans.len() + 1;
+        spans.push(Span {
+            label: attributes.metadata().name().to_string(),
+            ref_count: 1,
+        });
+        span::Id::from_u64(id as u64)
+    }
+
+    fn record(&self, span: &span::Id, _values: &span::Record) {
+        // Not entirely sure when this is called
```

Comment:
I'm not really sure when this is called, since everything seems to be recorded via events. Is events supposed to passthrough to record?